File: Python Excel Examples/RangesApiDemo/setRangeOutlineBorder.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.ranges_api.RangesApi(configuration)

name = "SetRangeOutlineBorder.xlsx"
folder = "Ranges"
storage = ""
sheetName = "Sheet1"
rangeOperate = spirecloudexcel.models.RangeSetOutlineBorderRequest()
rangeOperate.border_edge = "outline" # Available values: outline, vertical, horizontal, edgeleft
rangeOperate.border_style = "Medium" # Available values: Medium, DashDot, MediumDashDotDot, Thin
rangeOperate.border_color = spirecloudexcel.models.Color(255, 255, 0, 0)
Range = spirecloudexcel.models.Range()
Range.first_row = 1
Range.first_column = 1
Range.row_count = 5
Range.column_count = 10
rangeOperate.range=Range
api.set_range_outline_border(name, sheetName, rangeOperate, storage = storage, folder = folder)

File: JavaScript Excel Examples/RangesApiDemo/setRangeOutlineBorder.js
(function (){
    var appId = "your id";
    var appKey = "your key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel = require('../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.RangesApi(apiClient);
    
    var name = "setRangeOutlineBorder.xlsx";
    var opts = { folder: "input" };
    var sheetName = "Sheet1";
    var rangeOperate = new SpirecloudExcel.RangeSetOutlineBorderRequest();
    //Available values: outline, vertical, horizontal, edgeleft
    rangeOperate.borderEdge = "outline";
    //Available values: Medium, DashDot, MediumDashDotDot, Thin
    rangeOperate.borderStyle = "Medium";
    rangeOperate.borderColor = new SpirecloudExcel.Color();
    rangeOperate.borderColor.A = 255;
    rangeOperate.borderColor.R = 255;
    rangeOperate.borderColor.G = 0;
    rangeOperate.borderColor.B = 0;
    var range = new SpirecloudExcel.Range();
    range.FirstRow = 1;
    range.FirstColumn = 1;
    range.ColumnCount = 10;
    range.RowCount = 5;
    rangeOperate.Range = range;
    instance.setRangeOutlineBorder(name, sheetName, rangeOperate, opts, 
      function (error, data, response) {
        if (error) throw error;
    });
})();